solid_storage/src/lib.rs

```diff
@@ -1,3 +1,45 @@
+extern crate zbox;
+
+use std::io::{Read, Seek, SeekFrom, Write};
+use zbox::{init_env, OpenOptions, RepoOpener};
+
+pub fn init() {
+    println!("init storage adapter");
+    init_env();
+}
+
+pub fn read_file(path: &str) -> String {
+    // create and open a repository
+    let mut repo = RepoOpener::new().create(true).open("file://./solid-fs", "").unwrap();
+
+    // create and open a file for writing
+    let mut file = OpenOptions::new().create(true).open(&mut repo, path).unwrap();
+    let mut content = String::new();
+
+    // read file content using std::io::Read trait
+    file.seek(SeekFrom::Start(0)).unwrap();
+    file.read_to_string(&mut content).unwrap();
+
+    // return file content as String
+    content
+}
+
+pub fn write_file(path: &str, content: &str) -> bool {
```

Comment:
function should return a `Result<(), Error>`.

Reply:
✅